packages/main/src/backend/import/proxyConfig.ts

```diff
@@ -0,0 +1,103 @@
+import http from 'http';
+import https from 'https';
+import { ProxyAgent } from 'proxy-agent';
+import logger from '/@/logging/logger';
+
+let originalHttpAgent: typeof http.globalAgent | null = null;
+let originalHttpsAgent: typeof https.globalAgent | null = null;
+let currentProxyAgent: ProxyAgent | null = null;
+
+/**
+ * Gets the proxy configuration from environment variables
+ * Returns the proxy URL if found, and the NO_PROXY setting for bypass rules
+ */
+export function getProxyConfiguration(): { proxyUrl?: string; noProxy?: string } {
+  const proxyUrl =
+    process.env.HTTPS_PROXY ??
+    process.env.https_proxy ??
+    process.env.HTTP_PROXY ??
+    process.env.http_proxy ??
+    process.env.ALL_PROXY ??
+    process.env.all_proxy;
+
+  const noProxy = process.env.NO_PROXY ?? process.env.no_proxy;
+
+  return { proxyUrl, noProxy };
+}
+
+/**
+ * Checks if proxy needs to be configured based on environment variables
+ */
+function isProxyNeeded(): boolean {
+  const { proxyUrl } = getProxyConfiguration();
+  return !!proxyUrl;
+}
+
+/**
+ * Initializes proxy configuration if needed
+ * Sets up global HTTP/HTTPS agents with proxy configuration
+ */
+export function initProxyIfNeeded(): void {
+  if (!isProxyNeeded()) {
+    return;
+  }
+
+  const { proxyUrl, noProxy } = getProxyConfiguration();
+
+  if (!proxyUrl) {
+    return;
+  }
+
+  try {
+    originalHttpAgent = http.globalAgent;
+    originalHttpsAgent = https.globalAgent;
+
+    currentProxyAgent = new ProxyAgent();
+    http.globalAgent = currentProxyAgent;
+    https.globalAgent = currentProxyAgent;
+
+    logger.log(`Using proxy: ${proxyUrl}${noProxy ? ` with NO_PROXY: ${noProxy}` : ''}`);
+  } catch (error) {
+    logger.warn('Failed to initialize proxy agent:', error);
```

Comment:
Do you want it to run if the user configured the proxy but the proxy failed?

The user expected it to run with Proxy, without throwing, it will mislead the user